### ISSUE TITLE
VAX780 UNIX/32V SYSTEM TIME RUNNING TWICE AS FAST

### ISSUE THREAD

**Issue by WILLIAMCORCORAN** [simh/simh#253]
The VAX780 simulator, when running UNIX 32V, system clock is running twice as fast:

NOTE: The testing of this issue is incomplete. This entry is mainly a stub reporting the issue whilst we develop and execute a suitable test plan. @markpizz has directed that I create this entry now with the understanding that I provide appropriate details on this issue.

Here are some preliminary issue facts:

Excepting for the fact that the clock runs twice as fast (real time), other aspects are normal.
Calling "sleep 10" will properly sleep for ten system seconds; however, the sleep only lasts about 5 seconds in real time.
This following parameters/modes have no effect on this issue and the system clock continues to run about twice as fast:
SET IDLE=32 
SET NOIDLE
SET ASYNC
SET NOASYNC
SINGLE USER MODE 
MULTI USER MODE

Again, I will be positing more information such as VERSION, CLOCK, TEST SCRIPTS, TEST CONDITIONS, etc. This is only a placeholder.

**Comment by markpizz** [comment]
I believe that,, except for Unix 32V, the clock tick on ALL operating systems which use the VAX780 and all the other VAX processors is 10ms (100 Hz).  The MicroVAX systems all have a built-in 100 Hz tick generator.  The VAX780 has an interval timer which counts microseconds and this is always programmed to produce the 10ms tick.  

Unix 32V is setting the clock tick size to 16667 microseconds (aka 60Hz).   The above observations strongly suggests that even though the tick size is set to 16667 usecs ticks are actually being delivered at maybe 100 Hz or maybe even 120 Hz.

The original VAX780's interval timer model entangles 100 Hz behaviors while trying to implement the microsecond intervals and the result is what we're seeing now.  Recent changes in the simh timer/clock support make trying to model this much easier and simpler.  I'm working on fixing it.

**Comment by WILLIAMCORCORAN** [comment]
Hello Mark,

Understood.  

Is there anything here I can provide to you to help you?

For example, I was thinking about posting a 32V image stripped down, that only boots into single user and execs a program that demonstrates the error.

This way, once you think you've fixed it, you can quickly run this image and confirm that it's working.

Please let me know if you would like me to create this trivial 32V test image for you that demonstrates the issue with a google drive link.  

Otherwise,  please confirm you don't need anything else put in the case at this time.  

Or, please tell me what you need and I will put it in the case. 


Truly,

Bill Corcoran

**Comment by markpizz** [comment]
Bill,

I've got a 32V image which starts things normally.  If, for fun, you want to play around and dig through the 32V kernel source, maybe you can find where the 60 Hz interval is being programmed and change it to 100 Hz.  This is completely for fun and not relevant to fixing or testing the original problem.

Mark

**Comment by WILLIAMCORCORAN** [comment]
Cool!


Truly,

Bill Corcoran


On Dec 9, 2015, at 10:53 AM, Mark Pizzolato <notifications@github.com<mailto:notifications@github.com>> wrote:


Bill,

I've got a 32V image which starts things normally. If, for fun, you want to play around and dig through the 32V kernel source, maybe you can find where the 60 Hz interval is being programmed and change it to 100 Hz. This is completely for fun and not relevant to fixing or testing the original problem.

Mark

**Comment by markpizz** [comment]
Bill,

Please test the latest code and close this issue if you are satisfied that things now work correctly.

Mark

**Comment by WILLIAMCORCORAN** [comment]
Hello Mark,

This email confirms that the system time is no longer running twice as fast on the latest SIMH code for the VAX780.  Next, this email confirms that I performed a regression test on the PDP11 SIMH simulator running UNIX v7m.  In all cases, on both systems (VAX780 and PDP11) the system time appears to be running correctly.

Below, you will find the “SHOW VERSION” that serves to confirm my testing was completed on the appropriate version of the SIMH codebase.

Finally, without more, this case is now closed and I will update Github accordingly.

Truly,

Bill Corcoran



VAX 11/780 simulator V4.0-0 Beta

Simulator Framework Capabilities:

64b data

64b addresses

Ethernet Packet transports:PCAP:NAT:UDP

Idle/Throttling support is available

Virtual Hard Disk (VHD) support

Asynchronous I/O support

FrontPanel API Version 1

Host Platform:

Compiler: GCC 5.2.0

Simulator Compiled: Dec 10 2015 at 16:41:22

Memory Access: Little Endian

Memory Pointer Size: 64 bits

Large File (>2GB) support

RegEx support for EXPECT commands

OS clock resolution: 1ms

Time taken by msleep(1): 2ms

OS: Darwin PDP-11-70.local 15.0.0 Darwin Kernel Version 15.0.0: Sat Sep 19 15:53:46 PDT 2015; root:xnu-3247.10.11~1/RELEASE_X86_64 x86_64


        git commit id: 54bec5c1
